### PR TITLE
Updates and adjustments for manual, especially rewriters section

### LINKED ISSUE #258: Inconsistencies between text and code in the `rewrite` docs
Noticed a few inconsistencies in https://symbolicutils.juliasymbolics.org/rewrite/ (I guess some more automatic simplifications were added since the docs were written)?

Screen Shot 2021-03-29 at 8 09 30 PM
`(2*w)+(2*w)` already got simplified to `4w`
Screen Shot 2021-03-29 at 8 10 52 PM
Should at least one of these have returned "odd terms"?
Screen Shot 2021-03-29 at 8 12 17 PM
Same thing as first case

### PR DESCRIPTION
**Pull request by TymonKilich**
Hi, reading manual and getting myself familiar with SymbolicUtils I've tried to update and fix it a bit. 😉 

- I've added some info about automatic simplification of  in index.md;
- Updated docstring in rule.jl to make sense considering current autosimplification rules;
- Updated and extended rewrite.md file with adjustments for autosimplification, a bit more of verbosity for new users and few more examples.

I've planned to add "walking the tree" section in rewriters, but this PR is already taking too long, so decided to put it off to some other PR.

Please let me know if this makes sense and whether some corrections are necessary.

This should close #258.

PS Please feel free to squash the commits on merge (if it happens), they're a bit untidy, but I didn't know whether any corrections are needed, so decided not to.

### CODE REVIEW

File: page/index.md [hunk 1]
```diff
@@ -65,13 +65,40 @@ Note however that they are not subtypes of these types!
...
+2w + 3w - 3z + α
+```
+
+and reorders
+
+```julia:creating2
+(z + w)*(α + β)
+```
+
+expressions of `Symbolic{<:Number}` (which includes `Sym{Real}`) when they are created. It also does constant elimination (including rational numbers)
```
Comment (shashi) [inline page/index.md:89]: "of type `Symbolic{<:Number}`"?

Comment (shashi) [reply page/index.md:89]: this sentence with code interspersed is fun.

File: page/index.md [hunk 2]
```diff
@@ -65,13 +65,40 @@ Note however that they are not subtypes of these types!
...
+
+It's worth remembering that the expression may be transformed with respect to the input when it's created.
+
 You can do basic arithmetic on symbols to get symbolic expressions:
 
 ```julia:expr
-expr1 = α*sin(w)^2 +  β*cos(z)^2
-expr2 = α*cos(z)^2 +  β*sin(w)^2
+expr1 = α*sin(w)^2 + β*cos(z)^2
+expr2 = α*cos(z)^2 + β*sin(w)^2
```
Comment (shashi) [inline page/index.md:101]: This sentence and example should come before your changes above...

File: page/rewrite.md [hunk 3]
```diff
@@ -13,37 +15,60 @@ using SymbolicUtils
 
 (w, z, α, β) # hide
 
-r1 = @rule ~x + ~x => 2 * (~x)
+r1 = @rule sin(2(~x)) => 2sin(~x)*cos(~x)
 
-r1(sin(1+z) + sin(1+z))
+r1(sin(2z))
 ```
```
Comment (shashi) [inline page/rewrite.md:21]: Nice :clap:

File: page/rewrite.md [hunk 4]
```diff
@@ -94,39 +154,63 @@ rewriters.
...
 
-It applied `r1`, but didn't get the opportunity to apply `r2`. So we need to apply the ruleset again on the result.
+Important feature of `Chain` is that it returns the expressiona instead of `nothing` if it doesn't change the expression
 
-```julia:rewrite7
-rset(rset_result)
+```julia:composing2
+Chain([@acrule sin(~x)^2 + cos(~x)^2 => 1])((cos(x) + sin(x))^2)
 ```
```
Comment (shashi) [inline page/rewrite.md:189]: Some really neat examples!!

File: page/rewrite.md [hunk 5]
```diff
@@ -94,39 +154,63 @@ rewriters.
...
+
+```julia:composing4
+using SymbolicUtils.Rewriters: RestartedChain
+
+rcas = RestartedChain([acpyid, sqexpand])
+
+rcas((cos(x) + sin(x))^2)
+```
+
+It restarts the chain after each successful application of the rule, so after `sqexpand` is hit it (re)starts again and successfully applies `acpyid` to resulting expression.
```
Comment (shashi) [inline page/rewrite.md:210]: "successful application of _a_ rule"

**Comment by shashi** [comment]
Left some minor comments but this is super nice! :tada:

**Comment by TymonKilich** [comment]
I agree with all comments and put them into input. 

Thanks for the encouraging comments and work on this package. 😃

**Comment by TymonKilich** [comment]
Ah! I also remember now, that after update with autosimplifications testset `Chain, Postwalk and Fixpoint` make a bit less sense than it did before, since ex is autosimplified 

```julia
@testset "Chain, Postwalk and Fixpoint" begin
    @syms w z α::Real β::Real

    r1 = @rule ~x + ~x => 2 * (~x)
    r2 = @acrule ~x * +(~~ys) => sum(map(y-> ~x * y,  ~~ys));

    rset = Rewriters.Postwalk(Rewriters.Chain([r2]))
    @test getdepth(rset) == typemax(Int)

    ex = 2 * (w + w + α + β)

    @eqtest rset(ex) == (((2 * w) + (2 * w)) + (2 * α)) + (2 * β)
    @eqtest Rewriters.Fixpoint(rset)(ex) == ((2 * (2 * w)) + (2 * α)) + (2 * β)
end
```
It still passes but doesn't check for what it used to check for. It should probably be updated with some new examples. Chain can probably use example from the manual, but the rest needs something new I think.

Maybe I'll get to that, when I get to write something about walking the tree in manual but no promises 😅

**Pull request merged**